homeassistant/data_entry_flow.py

```diff
@@ -908,6 +922,10 @@ def async_set_progress_task(
         """Set in progress task."""
         self.__progress_task = progress_task
 
+    def is_matching(self, other_flow: Self) -> bool:
```

Comment:
`__eq__` is meant for this purpose I believe.

Reply:
Yes, you're right. I just thought it seemed ambitious to claim `__eq__` for this purpose.

If we use `__eq__`, do you suggest to immediately let `FlowHandler` override `__eq__` with some default implementation, which for example compares the `unique_id`, or do we leave the default, i.e. an identity check?

Also,`__eq__` allows passing in any object when comparing, but in this PR we only want to compare to other `FlowHandler` of the same class.

Reply:
Implementing `__eq__` interferes with hashing:
```
>>> class Test:
...     pass
...
>>> {Test()}
{<__main__.Test object at 0x7fad264978c0>}
>>> class Test2:
...     def __eq__(self, other):
...         return False
...
>>> {Test2()}
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unhashable type: 'Test2'
```

Reply:
Ok.